datadog/resource_datadog_screenboard.go

```diff
@@ -582,6 +588,25 @@ func resourceDatadogScreenboard() *schema.Resource {
 // # Convenience functions to safely pass info from Terraform to the Datadog API wrapper #
 // #######################################################################################
 
+func getMetadataFromJSON(jsonBytes []byte, unmarshalled interface{}) error {
+	decoder := json.NewDecoder(bytes.NewReader(jsonBytes))
+	// make sure we return errors on attributes that we don't expect in metadata
+	decoder.DisallowUnknownFields()
+	err := decoder.Decode(unmarshalled)
+	if err != nil {
+		return fmt.Errorf("Failed to unmarshal metadata_json: %s", err)
+	}
+	return nil
+}
+
+func validateMetadataJSON(v interface{}, k string) (ws []string, errors []error) {
+	err := getMetadataFromJSON([]byte(v.(string)), &map[string]datadog.TileDefMetadata{})
+	if err != nil {
+		errors = append(errors, fmt.Errorf("%q contains an invalid JSON: %s", k, err))
```

Comment:
Should the append be reversed here? The named return seems to want `string, errors`, wouldn't this return `errors, string`?

Reply:
This line actually only appends the current error to the slice of errors. I guess internally TF passes the `errors` slice around multiple validators and similar functions to collect all errors.

Reply:
Ahh ok, thanks 🙇